routing/providerquerymanager/providerquerymanager.go

```diff
@@ -504,6 +507,7 @@ func (npqm *newProvideQueryMessage) handle(pqm *ProviderQueryManager) {
 		}
 	} else {
 		trace.SpanFromContext(npqm.ctx).AddEvent("JoinQuery", trace.WithAttributes(attribute.Stringer("cid", npqm.k)))
+		log.Debugf("Joined existing query for cid %s which now has %d queries in progress", npqm.k, len(requestStatus.listeners)+1)
```

Comment:
not sure if meaningful, but maybe wrap it in `if log.Level().Enabled(zapcore.DebugLevel) {` so len is not calculated?